x-pack/plugins/security_solution/public/cases/components/connectors/servicenow/servicenow_sir_case_fields.test.tsx

```diff
@@ -194,5 +198,22 @@ describe('ServiceNowSIR Fields', () => {
         });
       })
     );
+
+    test('it should set subcategory to null when changing category', async () => {
+      await waitFor(() => {
+        const select = wrapper.find(EuiSelect).filter(`[data-test-subj="categorySelect"]`)!;
+        select.prop('onChange')!({
+          target: {
+            value: 'network',
+          },
+        } as React.ChangeEvent<HTMLSelectElement>);
+      });
+      wrapper.update();
+      expect(onChange).toHaveBeenCalledWith({
+        ...fields,
+        subcategory: null,
+        category: 'network',
+      });
+    });
```

Comment:
nit: I know the other tests here dont follow this and it passes as is without, but the correct way would be:
```suggestion
    test('it should set subcategory to null when changing category', async () => {
      const select = wrapper.find(EuiSelect).filter(`[data-test-subj="categorySelect"]`)!;
      select.prop('onChange')!({
        target: {
          value: 'network',
        },
      } as React.ChangeEvent<HTMLSelectElement>);
      wrapper.update();
      await waitFor(() => {
        expect(onChange).toHaveBeenCalledWith({
          ...fields,
          subcategory: null,
          category: 'network',
        });
      });
    });
```

Reply:
the await waitFor should be surrounding the expect after the async change. it definitely does not always work this way, but alas we're using enzyme with react-testing-library